pvlib/iotools/solcast.py

```diff
@@ -0,0 +1,459 @@
+    >>> )
+
+    use ``map_variables=False`` to avoid converting the data
+    to PVLib's conventions.
+
+    >>> df, meta = pvlib.iotools.solcast.get_solcast_live(
+    >>>     latitude=-33.856784,
+    >>>     longitude=151.215297,
+    >>>     map_variables=False,
+    >>>     api_key="your-key"
+    >>> )
+
+    References
+    ----------
+    .. [1] `Get an API Key <https://toolkit.solcast.com.au/register>`_
+    .. [2] `Solcast API Docs <https://docs.solcast.com.au/>`_
+    """
+
+    params = dict(
+        latitude=latitude,
+        longitude=longitude,
+        format="json",
+        **kwargs
+    )
+
+    data = _get_solcast(
+        endpoint="live/radiation_and_weather",
+        params=params,
+        api_key=api_key,
+        map_variables=map_variables
+    )
+
+    return data, {"latitude": latitude, "longitude": longitude}
+
+
+def _solcast2pvlib(data):
+    """Formats the data from Solcast to PVLib's conventions.
+
+    Parameters
+    ----------
+    data : pandas.DataFrame
+        contains the data as returned from the Solcast API
+
+    Returns
+    -------
+    a pandas.DataFrame with the data cast to PVLib's conventions
+    """
+    # move from period_end to period_middle as per pvlib convention
+    # to support Pandas 0.25 we cast PTXX to XX as ISO8601
+    # durations without days aren't supported:
+    # https://github.com/pandas-dev/pandas/pull/37159\
+    # Can remove once minimum supported Pandas version is >=1.2
+    periods = data.period.str.replace("PT", "").str.replace("M", "m")
+
+    data["period_mid"] = pd.to_datetime(
+        data.period_end) - pd.to_timedelta(periods) / 2
+    data = data.set_index("period_mid").drop(columns=["period_end", "period"])
+
+    # rename and convert variables
+    for variable in VARIABLE_MAP:
+        if variable.solcast_name in data.columns:
+            data.rename(
+                columns={variable.solcast_name: variable.pvlib_name},
+                inplace=True
+            )
+            data[variable.pvlib_name] = data[
+                variable.pvlib_name].apply(variable.conversion)
+    return data
+
+
+def _get_solcast(
+        endpoint,
+        params,
+        api_key,
+        map_variables
+):
+    """retrieves weather, irradiance and power data from the Solcast API
+
+    Parameters
+    ----------
+    endpoint : str
+        one of Solcast API endpoint:
+            - live/radiation_and_weather
+            - forecast/radiation_and_weather
+            - historic/radiation_and_weather
+            - tmy/radiation_and_weather
+    params : dict
+        parameters to be passed to the API
+    api_key : str
+        To access Solcast data you will need an API key [1]_.
+    map_variables: bool, default: True
+        When true, renames columns of the DataFrame to PVLib's variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+        Time is the index as midpoint of each interval
+        from Solcast's "period end" convention.

```

Comment:
Right now the documentation of the child functions ``map_variables`` parameter only states:
>  When true, renames columns of the DataFrame to pvlib variable names where applicable

I think that all instance of ``map_variables`` should also denote that the timestamp is shifted (I don't recall seeing this anywhere else in pvlib, but do support the idea).